app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt

```diff
@@ -0,0 +1,73 @@
+package org.p2p.wallet.rpc.interactor
+
+import org.p2p.solanaj.core.Account
+import org.p2p.solanaj.core.Transaction
+import org.p2p.solanaj.programs.TokenProgram
+import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
+import org.p2p.wallet.rpc.repository.RpcRepository
+import org.p2p.wallet.utils.toPublicKey
+
+class TokenInteractor(
+    private val rpcRepository: RpcRepository,
+    private val addressInteractor: TransactionAddressInteractor,
+    private val tokenKeyProvider: TokenKeyProvider
+) {
+
+    suspend fun close(addressToClose: String): String {
+        val owner = tokenKeyProvider.publicKey.toPublicKey()
+        val instruction = TokenProgram.closeAccountInstruction(
+            TokenProgram.PROGRAM_ID,
+            addressToClose.toPublicKey(),
+            owner,
```

Comment:
named arg is better here

Reply:
it's java class which does not support named args